packages/pros/src/adi/port.rs

```diff
@@ -0,0 +1,83 @@
+
+use core::{
+    ffi::c_int,
+    ops::{Deref, DerefMut},
+};
+
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use crate::error::bail_on;
+
+use pros_sys::{PROS_ERR, adi_port_config_e_t, E_ADI_DIGITAL_IN, E_ADI_ANALOG_OUT, E_ADI_DIGITAL_OUT, E_ADI_ANALOG_IN, E_ADI_LEGACY_ENCODER, E_ADI_LEGACY_ULTRASONIC };
+
+pub struct AdiPort(u8);
+
+impl AdiPort {
+    /// Create an AdiPort, returning err `AdiError::InvalidPort` if the port is invalid.
```

Comment:
Thsi comment seems incorrect, since this type returns `Option<Self>`, although it might be better to just remove the unreachable check and always return `Self`.